### PR TITLE
[PVR][guiinfo] Recordings: Add more VIDEOPLAYER info labels (e.g. VIDEOPLAYER_EPISODENAME).

### PR DESCRIPTION
**Pull request by ksooo**
This makes it possible for skinners to get more data for PVR recordings, for instance VIDEOPLAYER_EPISODENAME now also works for PVR recordings.

@Jalle19 for review?
@phil65 @BigNoid @HitcherUK fyi
@HitcherUK i will come up with a PR for confluence for recording episode name display

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -8674,10 +8674,91 @@ std::string CGUIInfoManager::GetVideoLabel(int item)
...
+      case VIDEOPLAYER_YEAR:
+        if (tag->HasYear())
+          return StringUtils::Format("%i", tag->GetYear());
+        break;
+
+      case VIDEOPLAYER_EPISODE:
+        if (tag->m_iEpisode > 0)
+        {
+          if (tag->m_iEpisode == 0) // prefix episode with 'S'
+            return StringUtils::Format("S%i", tag->m_iEpisode);
```
Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: Should probably be `E`, not `S`?

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -8674,10 +8674,91 @@ std::string CGUIInfoManager::GetVideoLabel(int item)
...
+          else
+            return StringUtils::Format("%i", tag->m_iEpisode);
+        }
+        break;
+
+      case VIDEOPLAYER_SEASON:
+        if (tag->m_iSeason > 0)
+          return StringUtils::Format("%i", tag->m_iSeason);
```
Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: If you prefix the episode, perhaps it would make sense to prefix the season too? Or prefix none and leave it up to skins to do the formatting.

File: xbmc/GUIInfoManager.cpp [hunk 3]
```diff
@@ -8674,10 +8674,91 @@ std::string CGUIInfoManager::GetVideoLabel(int item)
...
+        break;
+
+      case VIDEOPLAYER_EPISODENAME:
+        return tag->EpisodeName();
+
+      case VIDEOPLAYER_CHANNEL_NAME:
+        return tag->m_strChannelName;
```
Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: Are these properties public or is this class a friend class? Not quite related to this PR but perhaps this could use some refactoring.

File: xbmc/GUIInfoManager.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: No discussion about this, please. It is exactly like it is done for VideoInfoTags etc. I once had a discussion about this with montellese and he was very clear that this has to be exactly like it is. ;-) Okay?

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: See my other comment, please. ;-)

File: xbmc/GUIInfoManager.cpp (hunk 3 above)
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: public. Just one of so many examples of not encapsulating data properly.

File: xbmc/GUIInfoManager.cpp [hunk 4]
```diff
@@ -8674,10 +8674,91 @@ std::string CGUIInfoManager::GetVideoLabel(int item)
...
+        const CPVRChannelPtr channel(tag->Channel());
+        if (channel)
+          return channel->FormattedChannelNumber();
+        break;
+      }
+
+      case VIDEOPLAYER_CHANNEL_GROUP:
+      {
+        if (!tag->IsRadio())
+          return g_PVRManager.GetPlayingTVGroupName();
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: No, we havn't. I have just copied the gist of the implementation of VIDEOPLAYER_CHANNEL_GROUP for CPVRChannel. I guess this can be improved, but wanted to do this in another PR.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: This can't be right, no one in their right mind would except "episode 4" to be abbreviated `S04`?

Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: I have to admit that this looks really strange. If it is actually a bug, it has to be fixed here and at least two other places in this file. The other places date back until 2013... I would prefer to do the fix in a separate PR. Okay?

Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: Of course

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
jenkins and travis errors are unrelated
